### PR TITLE
PR: Update minimal required version of `pylint` to `>=2.5.0,<3.1`

### PR DESCRIPTION
**Pull request by bnavigator**
## Description of Changes

* [ ] Wrote at least one-line docstrings (for any new functions)
* [ ] Added unit test(s) covering the changes (if testable)

* [ ] Included a screenshot or animation (if affecting the UI, see [Licecap](https://www.cockos.com/licecap/))

Pylint 3 removed long deprecated `find_pylintrc()` which was only a wrapper to `find_default_config_files()`.
https://github.com/pylint-dev/pylint/issues/9105

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct:
bnavigator

### CODE REVIEW

**Comment by dalthviz** [comment]
Note: If we decide to include this on 5.5.0, this will need to be rebased into the `5.x` branch

**Comment by ccordoba12** [comment]
Agreed.

@bnavigator, to include this in our next release, please run the following commands locally:

    git checkout 5.x
    git checkout pylint3
    git rebase --onto 5.x master pylint3
    git push -f origin pylint3

**Comment by ccordoba12** [comment]
Note: This requires to merge first PR https://github.com/python-lsp/python-lsp-server/pull/475 and then sync our subrepo here so that our tests actually run with Pylint 3 and catch any possible regression.

**Comment by ccordoba12** [comment]
@bnavigator, I'm going to take care of finishing this one so we can be sure to include it in 5.5.

File: .github/scripts/install.sh [hunk 1]
```diff
@@ -26,6 +26,9 @@ if [ "$USE_CONDA" = "true" ]; then
     # Remove pylsp before installing its subrepo below
     micromamba remove --force python-lsp-server python-lsp-server-base -y
 
+    # Install Pylint 3 to run our tests with it
+    micromamba install pylint=3 -q -y
```
Comment (ccordoba12) [inline .github/scripts/install.sh:30]: Remove these lines when PyLSP 1.9 is released.

File: .github/scripts/install.sh [hunk 2]
```diff
@@ -42,6 +45,9 @@ else
     # To check our manifest
     pip install -q check-manifest
 
+    # Install Pylint 3 to run our tests with it
+    pip install -U pylint
```
Comment (ccordoba12) [inline .github/scripts/install.sh:49]: And these lines too.

**Review by ccordoba12 — COMMENTED** [review]
@dalthviz, these are a couple of comments for you before doing the release.

**Review by ccordoba12 — APPROVED** [review]
Thanks for your help with this @bnavigator!

**Pull request merged**